File: ruby/pangram/pangram.rb
require 'pry'
class Pangram

  def self.pangram?(phrase)
    binding.pry
    current = phrase.split('').uniq.join.gsub('/[^a-z]/i', '')
    p current
    p current.length
    evaluate_result(current)
  end

  private

    def self.evaluate_result(phrase)
      if phrase.length == 26
         true
      else
        false
      end
    end
end

File: javascript/pangram/pangram.js
var pry = require('pryjs');
var pangram = function(string){
  this.string = string
};

pangram.prototype.isPangram = function() {
  var cleanedChars = cleanChars(getUnique(this.string))
  eval(pry.it);
  if(cleanedChars.length === 26) {
    return true
  } else {
    return false
  }
};

function getUnique(string){
  var unique = string.toLowerCase().split('').filter(function(elem, index, self) {
    return index == self.indexOf(elem);
  })
  return unique.join('')
}

function cleanChars(string){
  var clean = string.replace(/[^a-z]/g, '');
  return clean
}

module.exports = pangram;
